### ISSUE TITLE
[feature]: Purchase History for Venia (PWA)

### ISSUE THREAD

**Issue by ericerway** [magento/pwa-studio#294]
## This issue is for the following packages:

[X] `venia-concept`
[ ] `pwa-buildpack`
[X] `peregrine`
[ ] `pwa-module`
[ ] `pwa-devdocs`

## Description

(link from Invision to Purchase History; related to account/auth work)

As a shopper, I want to find previous purchases (by item) on Venia by order# so that I can make a repeat purchase or request help from Customer Service.

#### Assumptions
- Requires authentication (e.g. Roni Cost) to access this data
- Grouping for Last 30 Days and Earlier (all-time) are desirable but stretch goals
- Order Details will be handled separately; this is just the listing for Purchase History
- Filtering will be considered in the future (backlog)
- Utilize REST API as GraphQL is in development (for orders)

## Tasks
- [ ] Discovery to retrieve orders data from REST or GraphQL (and auth)
- [ ] Identify future considerations (tasks) for filtering, pagination and other experiences
- [ ] Implement basic Purchase History (last orders; items)
- [ ] Testing and documentation (as appropriate)

**Comment by dmtrkad** [comment]
**Q&A**
1. Mockups need to be updated 
2. What is the key factor of putting a common component into Venia concept package or Peregrine package?

**Comment by ericerway** [comment]
- Mockups available on Invision via @soumya-ashok here: https://magento.invisionapp.com/share/WQN5F7BYBPG
- @zetlen can you help on the Venia concept vs. Peregrine?

**Comment by zetlen** [comment]
@dmtrkad Initially, we do all our development in Venia. When we feel personally we have created a component or utility that is useful outside Venia, we can PR it to peregrine.

**Comment by Starotitorov** [comment]
Frontend tasks:
- [ ] Implement page component.
- [ ] Implement filter component.
- [ ] Implement purchase history list component.
- [ ] Implement fetching of purchase history (last orders; items).
- [ ] Add unit tests.
- [ ] Add docs.

**Comment by vitalics** [comment]
Scenario is ready for review and available from the link below
https://app.hiptest.com/l/sc/2338900

**Comment by AlexeyKaryka** [comment]
@ericerway @zetlen should we show duplicated items from different orders?

**Comment by ericerway** [comment]
@dmtrkad this is only for authenticated users; order details is for both authenticated and guest users. Let's create a small user story for the remaining 1SP work here. 

@AlexeyKaryka yes let's keep duplicated items; this should be an uncommon use case. Thanks!

**Comment by ericerway** [comment]
Double checked and the endpoint is definitely there for order details *but* may be for admin only and will not work for cookie auth. Let's handle offline but please double check the new REST API documentation here: https://devdocs.magento.com/redoc/2.3/index.html

**Comment by Starotitorov** [comment]
@ericerway, we user header Authorization to attach token to request, we do not user cookies, we store this token in localStorage. It is possible to retreive order information only for authorized users. We do not have some kind of authorization token for guest users.

**Comment by Starotitorov** [comment]
@zetlen, I faced with an issue, I did not find an endpoint to fetch all items from all orders for customer or at least endpoint to fetch all orders for customer using auth token https://devdocs.magento.com/redoc/2.3/ https://devdocs.magento.com/guides/v2.3/rest/list.html . Do we have appropriate endpoint in general? I see that we have an endpoint /V1/orders, but it is for admins.

**Comment by Starotitorov** [comment]
@ericerway, @awilcoxa, @zetlen, @dmtrkad We are blocked by issue related to running units tests and importing List from peregrine, there is a pull request https://github.com/magento-research/pwa-studio/pull/417. Also, there is an open question related to rest API endpoint that can be used to fetch information about items from all orders that were submitted by authorized customer https://github.com/magento-research/pwa-studio/issues/294#issuecomment-433092904.

**Comment by soumya-ashok** [comment]
![purchase_details_Screen.PNG](https://images.zenhubusercontent.com/5b89562fba905a6006740695/dd83c9e0-9509-418d-916a-33c71d708e43)

Based on the mockup sent via email, here are my notes -
The font on the screen does not seem to be 'Muli'
The filter icon needs to be reduced in size, and vertically aligned in the field
How would the filtering interaction work? Do they enter a filter and select "done" on the keypad screen overlay, and the page reloads with results or
Do the results filter automatically when they complete the filter term in the field? @jimbo could you assist here please?

**Comment by Starotitorov** [comment]
@soumya-ashok, thank you for the feedback! According to requirements filtering is out of the scope of the story and will be implemented in the future. Right now filter field is just a static component.

**Comment by Starotitorov** [comment]
@soumya-ashok, @ddzhura, @jimbo

I will align filter icon vertically and reduce the size, thank you for the comment.


- We specified 'Muli' font, but here a fallback font is used, because we do not load 'Muli' font in our application.
- Our icon library https://feathericons.com/ does not contain filter icon from mockup, so we added a similar icon. Do we need to change this?

**Comment by soumya-ashok** [comment]
@Starotitorov - The mockups use material icons, but @jimbo and I decided to go with Feather icons because they are open source and suit the design a little better. The feather filter icon is fine the way you have it right now unless there is a "filled" option available in the same library.

**Comment by Starotitorov** [comment]
@soumya-ashok, there is no problem to change a color of the icon, yeah, i will do it.

**Comment by ericerway** [comment]
Moving to Sprint 30 for review while fetch endpoints are explored.

**Comment by awilcoxa** [comment]
@jimbo is purchase history merged/released? Anything left to do for this issue?

**Comment by awilcoxa** [comment]
Components exist but is not available in UI, moving back to backlog to determine next steps/stories.

**Comment by awilcoxa** [comment]
https://jira.corp.magento.com/browse/PWA-276

**Comment by sivakumarkoduru** [comment]
> ![purchase_details_Screen.PNG](https://camo.githubusercontent.com/355ef5141d40d662c4758e01a65132053c63b23f/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3562383935363266626139303561363030363734303639352f64643833633965302d393530392d343138642d393136612d333363373164373038653433)
> 
> @ddzhura
> Based on the mockup sent via email, here are my notes -
> The font on the screen does not match the mockup. 'Muli' is the one we are using. It is a free Google font.
> The filter icon needs to be reduced in size, and vertically aligned in the field
> How would the filtering interaction work? Do they enter a filter and select "done" on the keypad screen overlay, and the page reloads with results or
> Do the results filter automatically when they complete the filter term in the field? @jimbo could you assist here please?

Its better to add item status in the grid ,for example delivered on 12th May 2019 , cancelled,returned etc,

**Comment by sirugh** [comment]
This is outdated. We have internal work to address the order history page now, starting with #2611
